device.cpp

```diff
@@ -2157,6 +2157,7 @@ Device::Device(DeviceKey key, deCONZ::ApsController *apsCtrl, QObject *parent) :
     addItem(DataTypeString, RAttrDdfPolicy);
     addItem(DataTypeString, RAttrDdfHash);
     addItem(DataTypeUInt32, RAttrOtaVersion);
+    addItem(DataTypeUInt16, RAttrZoneType);
```

Comment:
Just noticed this one, I'm not too happy about adding `RAttrZoneType`  in this place and to every device.

I think it's okay to get the PR up and running, but ideally we can mark items in sub-resources to be cascaded to the parent (aka Device*) in future.

Reply:
I think there's a bit of a catch 22 there?  We only have the subdevices once the DDF has been matched, but we need the zone type to match the DDF.  We might do a check that it gets added only to devices with the _IAS Zone_ cluster?

Reply:
Unfortunately, this doesn't work.  I think the value of `RAttrZoneType` is not persisted across restarts.  After re-adding the logic to persist zone type in `ZCL_VALUES`, it still doesn't work, as the attempted restore happens before the state enter event.  When re-creating the item on restore, it works!

> but there's no need to save its value explicitly to `ZCL_VALUES`

I can no longer reproduce this.  Maybe it worked, because at that time the resource item was still defined on the subdevices, or simply because the value was still in ZCL_VALUES.  Or maybe I simply forgot to deploy the plugin after compiling.  When manually removing the value from the database, the `sensors` resource reverts to ZHAAlarm.